### PR TITLE
EZP-26794 : unable to login in backend

### PR DESCRIPTION
**Pull request by nicolas-bastien**
Jira : https://jira.ez.no/browse/EZP-26794

## Description :

Clear unauthorized user authentication if they don't have acces to remove siteaccess unwanted exception

## Test manually

Trying to login using a user whose user group has no assigned role
- Create a user group - Group1
- Under Group1, create a user - user1 _(don't assign any role to Group1)_
- Try to login in admin interface using the user1
- We are unable to do so, because of a "401 - unauthorized" - ok once again
- Right after, try to login as admin

-> i should work

## QA 

    composer require ezsystems/ezpublish-kernel:dev-EZP-26794-unable-to-login

### CODE REVIEW

File: eZ/Publish/Core/REST/Server/Controller/SessionController.php [hunk 1]
```diff
@@ -92,8 +92,10 @@ public function createSessionAction(Request $request)
             // Already logged in with another user, this will be converted to HTTP status 409
             return new Values\Conflict();
         } catch (AuthenticationException $e) {
+            $this->authenticator->logout($request);
```
Comment (bdunogier) [inline eZ/Publish/Core/REST/Server/Controller/SessionController.php]: Is the user authenticated when there is an authentication error ? I understand that the later case, where the user doesn't have enough permissions, leaves the user logged in, but I don't think it does, or should, when auth failed.

Comment (nicolas-bastien) [inline eZ/Publish/Core/REST/Server/Controller/SessionController.php]: @bdunogier you're right, done

**Comment by andrerom** [comment]
Is this fixing two issues? I don't fully see the connection with the logout() call to making use `ez` route does not throw a `UnauthorizedSiteAccessException`.

**Comment by bdunogier** [comment]
The idea is that if the account doesn't have the right authorizations, then the user shouldn't be logged in, as it will be prevent him from trying to login again.

But in retrospect, I am not sure what the route check on `/ez` or not.

**Comment by andrerom** [comment]
and by "not right authorizations", we are talking about not access to user/login right? or?

**Comment by bdunogier** [comment]
> we are talking about not access to user/login right? or?

Yes sir.

**Comment by andrerom** [comment]
Then I'm +1
But what are your concern regarding the `ez` route check?

**Comment by nicolas-bastien** [comment]
@andrerom you're right the ez route check was not needed

So here is my PR updated a single line ! 

/cc @bdunogier

**Comment by bdunogier** [comment]
@nicolas-bastien you can send this to QA now (and _then_ merge ;) )

**Comment by nicolas-bastien** [comment]
I just did it and I'll wait this time ;-)

**Pull request merged**